File: libs/eave-stdlib-py/src/eave/stdlib/core_api/models/documents.py
from typing import Optional
from eave.stdlib.core_api.models import BaseResponseModel
from eave.stdlib.core_api.models import BaseInputModel


class DocumentSearchResult(BaseResponseModel):
    title: str
    url: str


class DocumentInput(BaseInputModel):
    title: str
    content: str
    parent: Optional["DocumentInput"] = None

File: libs/eave-stdlib-ts/src/core-api/models/documents.ts
export type DocumentInput = {
  title: string;
  content: string;
  parent: DocumentInput | null;
}

export type EaveDocument = {
  title: string;
  content: string;
  parent: EaveDocument | null;
}

export type DocumentSearchResult = {
    title: string;
    url: string;
}
